localstack/services/lambda_/provider.py

```diff
@@ -1678,7 +1678,12 @@ def update_event_source_mapping(
         context: RequestContext,
         request: UpdateEventSourceMappingRequest,
     ) -> EventSourceMappingConfiguration:
-        state = lambda_stores[context.account_id][context.region]
+        account_id, region = api_utils.get_account_and_region(request["FunctionName"], context)
```

Comment:
💡 Is that the new pattern we should prefer over getting the `account_id` and `region` directly from the context?

Reply:
It looks like `FunctionName` does not always exist (3 failing tests with KeyError):

```
E             File "/opt/code/localstack/localstack/services/lambda_/provider.py", line 1681, in update_event_source_mapping
E               account_id, region = api_utils.get_account_and_region(request["FunctionName"], context)
E                                                                     ~~~~~~~^^^^^^^^^^^^^^^^
E           KeyError: 'FunctionName'
```